tools/topology/sof/pipe-volume-demux-playback.m4

```diff
@@ -22,8 +22,29 @@ include(`muxdemux.m4')
 include(`mixercontrol.m4')
 include(`bytecontrol.m4')
 
-# Use default parameters
-include(`demux_coef_default.m4')
+dnl pipeline_id, channels, routing_matrix_rows
+define(matrix1, `ROUTE_MATRIX(PIPELINE_ID, 2,
+			     `BITS_TO_BYTE(1, 0, 0 ,0 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 1, 0 ,0 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 1 ,0 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,1 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,1 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,1 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,0 ,1 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,0 ,0 ,1)')')
+
+define(matrix2, `ROUTE_MATRIX(5, 2,
+			     `BITS_TO_BYTE(1, 0, 0 ,0 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 1, 0 ,0 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 1 ,0 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,1 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,1 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,1 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,0 ,1 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,0 ,0 ,1)')')
+
+dnl frame_format, num_channels, num_streams, route_matrix
+MUXDEMUX_CONFIG(demux_priv, 2, 2, 2, LIST(`	', `matrix1,', `matrix2'))
```

Comment:
```suggestion
MUXDEMUX_CONFIG(demux_priv, FMT(PIPELINE_FORMAT), PIPELINE_CHANNELS, 2, LIST(`	', `matrix1,', `matrix2'))
```
you will need a function to convert the format in you define in mux.h

Reply:
I think artur's PR is also removing this field also. We should have all pipeline params come from parameter propagation in pipelines. Only valid field would be the number of streams.